src/main/java/org/openhab/tools/analysis/checkstyle/api/AbstractStaticCheck.java

```diff
@@ -77,81 +80,73 @@ protected int findLineNumber(String[] fileContent, String searchedText, int star
             return info;
         } catch (IOException e) {
-            throw new CheckstyleException("Unable to read from file: " + file.getAbsolutePath(), e);
+            throw new CheckstyleException("Unable to read from file: " + fileText.getFile().getAbsolutePath(), e);
         } catch (ParseException e) {
-            throw new CheckstyleException("Unable to parse file:" + file.getAbsolutePath(), e);
+            throw new CheckstyleException("Unable to parse file:" + fileText.getFile().getAbsolutePath(), e);
         }
     }
 
     /**
      * Reads a properties list from a file
      *
-     * @param file - the input file
+     * @param fileText - Represents the text contents of a file
      * @return Properties object containing all the read properties
      * @throws CheckstyleException - if an error occurred while trying to parse the file
      */
-    protected Properties readPropertiesFromFile(File file) throws CheckstyleException {
-        String filePath = file.getAbsolutePath();
+    protected Properties readPropertiesFromFile(FileText fileText) throws CheckstyleException {
 
-        try (InputStream buildPropertiesInputStream = new FileInputStream(filePath)) {
+        try {
             Properties properties = new Properties();
-            properties.load(buildPropertiesInputStream);
+            properties.load(getInputStream(fileText));
             return properties;
         } catch (FileNotFoundException e) {
-            throw new CheckstyleException("File: " + file.getAbsolutePath() + " does not exist.", e);
+            throw new CheckstyleException("File: " + fileText.getFile().getAbsolutePath() + " does not exist.", e);
         } catch (IOException e) {
-            throw new CheckstyleException("Unable to read properties from: " + file.getAbsolutePath(), e);
+            throw new CheckstyleException("Unable to read properties from: " + fileText.getFile().getAbsolutePath(), e);
         }
     }
 
     /**
      * Parses the content of a given file as a HTML file
      *
-     * @param file - the input file
+     * @param fileText - Represents the text contents of a file
      * @return HTML Document representation of the file
-     * @throws CheckstyleException - if an error occurred while trying to parse the file
      */
-    protected org.jsoup.nodes.Document parseHTMLDocumentFromFile(File file) throws CheckstyleException {
-        try {
-            byte[] fileByteArray = Files.readAllBytes(file.toPath());
-            String fileContent = new String(fileByteArray);
-            org.jsoup.nodes.Document fileDocument = Jsoup.parse(fileContent);
-            return fileDocument;
-        } catch (IOException e) {
-            throw new CheckstyleException("Unable to read the content of the file" + file.getAbsolutePath(), e);
-        }
+    protected org.jsoup.nodes.Document parseHTMLDocumentFromFile(FileText fileText) {
```

Comment:
Good moment to properly import (if possible) and do no store the local before returning?

Reply:
Unfortunately there is already an org.w3c.dom.Document imported.